integration-tests/contract-security-wrong-field-annotations.spec.ts

```diff
@@ -0,0 +1,275 @@
+      // Calling Michelson code: CONTRACT %add (pair nat nat) ;
+
+      const op4 = await Tezos.contract
+        .at(contractAddition.address)
+        .then((contract) => {
+          return contract.methodsObject
+            .callNoAnnot({
+              entrypoint: 'Add',
+              param: {
+                valueA: `1`,
+                valueB: `2`,
+                1: `1`,
+                2: `2`,
+              },
+            })
+            .send();
+        })
+        .then((op4) => {
+          return op4.confirmation().then(() => op4.hash);
+        });
+
+      //Testcase #5
+      // Calls entrypoint: CONTRACT %addNoAnnot (pair nat nat) ;
+      // Calling Michelson code: CONTRACT %addNoAnnot (pair (nat %valueA) (nat %valueB)) ;
+
+      const op5 = await Tezos.contract
+        .at(contractAddition.address)
+        .then((contract) => {
+          return contract.methodsObject
+            .callNoAnnot({
+              entrypoint: 'AddNoAnnot',
+              param: {
+                1: `1`,
+                2: `2`,
+              },
+            })
+            .send();
+        })
+        .then((op5) => {
+          return op5.confirmation().then(() => op5.hash);
+        });
+
+      //Testcase #6
+      // Calls entrypoint: CONTRACT %addWrongAnnot (pair nat nat) ;
+      // Calling Michelson code: CONTRACT %addWrongAnnot (pair (nat %valueA) (nat %valueB)) ;
+
+      const op6 = await Tezos.contract
+        .at(contractAddition.address)
+        .then((contract) => {
+          return contract.methodsObject
+            .callNoAnnot({
+              entrypoint: 'AddWrongAnnot',
+              param: {
+                1: `1`,
+                2: `2`,
+              },
+            })
+            .send();
+        })
+        .then((op6) => {
+          return op6.confirmation().then(() => op6.hash);
+        });
+
+      //Testcase #7
+      // Calls entrypoint: (pair %add (nat %valueA) (nat %valueB))
+      // Calling Michelson code: CONTRACT %add (pair (nat %a) (nat %b)) ;
+
+       const op7 = await Tezos.contract
+         .at(contractAddition.address)
+         .then((contract) => {
+           return contract.methodsObject
+             .callWrongAnnot({
+               entrypoint: 'Add',
+               param: {
+                 a: `1`,
+                 b: `2`,
+               },
+             })
+             .send();
+         })
+         .then((op7) => {
+           return op7.confirmation().then(() => op7.hash);
+         });
+
+      //Testcase #8
+      // Calls entrypoint: (pair %addNoAnnot nat nat))
+      // Calling Michelson code: CONTRACT %add (pair (nat %a) (nat %b))
+
+      try{
+      const op = await Tezos.contract
+        .at(contractAddition.address)
+        .then((contract) => {
+          return contract.methodsObject
+            .callWrongAnnot({
+              entrypoint: 'AddNoAnnot',
+              param: {
+                1: `1`,
```

Comment:
Instead of 1 , the annot should be a
Instead of 2, the annot should be b

Reply:
I made the change but it still gives the same error,.